src/test/java/GmailClientTest.java

```diff
@@ -130,15 +140,54 @@ public void getQueryStringDontFilterForImportantTag() {
   }
 
   @Test
-  public void getQueryStringCombined() {
-    // Should return query matching all specified rules
-    // trailing spaces on sub-queries do not matter for this method, so they will not be considered
-    String multipleFilterQuery =
-        GmailClient.emailQueryString(ONE_UNIT_OF_TIME, DAYS_UNIT, true, true, SAMPLE_EMAIL);
-
-    Assert.assertTrue(multipleFilterQuery.contains(ONE_DAY_QUERY));
-    Assert.assertTrue(multipleFilterQuery.contains(UNREAD_EMAILS_QUERY));
-    Assert.assertTrue(multipleFilterQuery.contains(IMPORTANT_EMAILS_QUERY));
-    Assert.assertTrue(multipleFilterQuery.contains(FROM_EMAIL_QUERY));
+  public void getQueryStringNoSubjectWords() {
+    String subjectLineQuery = GmailClient.wordsInSubjectLineQuery(EMPTY_SUBJECT_LINE_WORDS);
+
+    Assert.assertEquals(subjectLineQuery, EMPTY_QUERY);
+  }
+
+  @Test
+  public void getQueryStringSomeSubjectWords() {
+    String subjectLineQuery = GmailClient.wordsInSubjectLineQuery(SOME_SUBJECT_LINE_WORDS);
+
+    // Query should be in the form of "subject:("wordOne" OR "wordTwo" OR "wordThree")"
+    // The order of the words in the brackets does not matter.
+    String actualSubjectLineQueryPrefix =
+        subjectLineQuery.substring(0, subjectLineQuery.indexOf(':') + 1);
+    List<String> actualSubjectLineWords =
+        Arrays.asList(
+            subjectLineQuery
+                .substring(subjectLineQuery.indexOf('(') + 1, subjectLineQuery.indexOf(')'))
+                .split(" OR "));
+
+    // Expected words should be the same as the input, with quotes surrounding each word (if not
+    // already present)
+    // Ex: "WordOne", wordTwo -> "WordOne", "wordTwo"
+    List<String> expectedSubjectLineWords =
+        SOME_SUBJECT_LINE_WORDS.stream()
+            .map((word) -> String.format("\"%s\"", StringUtils.strip(word, "\"")))
+            .collect(Collectors.toList());
```

Comment:
You don't need to recreate this logic in the test. You are providing the input data, so you know if the individual tokens are quoted or not. You should just have a separate test case to check that quotes are handled correctly.

In general, we want to make tests as simple as possible. We don't write tests for tests, so it's hard to verify correctness if the test logic is not trivial.

Reply:
Agreed - I'm just going to make the assumption here that the order of the words should be the same in the string (while they don't have to be, I don't think this is a bad assumption - there would be no reason to change the order) so I can compare the strings exactly.